src/Umbraco.Infrastructure/Security/MemberRoleStore.cs

```diff
@@ -0,0 +1,342 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Core.Security
+{
+    /// <summary>
+    /// A custom user store that uses Umbraco member data
+    /// </summary>
+    public class MemberRoleStore<TRole> : IRoleStore<TRole> where TRole : IdentityRole
+    {
+        private readonly IMemberGroupService _memberGroupService;
+        private bool _disposed;
+
+        //TODO: Move into custom error describer.
```

Comment:
What is our preferred error message approach?

Reply:
I think we'll need to revisit errors from our identity implementation since that isn't a thing in v8. We probably have some of these errors already translated in our language files/service. I think we'll need to create a separate task for dealing with that. (i'll create one)

Reply:
Ah yes, makes sense as a separate task.